scripts/workdir-cd

```diff
@@ -800,64 +797,51 @@ rcd-set_repopath() {
 	split paths "${O[path]}" ':'
 
 	local path
+	local found_default="false"
 	for path in "${paths[@]}"; do
 		[[ ${path} ]] || continue
 
 		if [[ ${path} == default ]]; then
-			# use q to avoid portageq's slow startup
-			unset DEBUG # avoid q noise if set
-			q -Cqo | while IFS= read -r path; do
-				# gentoo: ${ROOT}/var/db/repos/gentoo (main)
-				path=${path% (*)}
-				path=${path#*: }
-
-				# q properly uses repos.conf from prefix but does not include
-				# the prefix in the output (just ROOT), so do hacky stuff
-				[[ ${EPREFIX} && ${path} != "${EROOT}/"* ]] &&
-					path=${EROOT}${path#"${ROOT}"}
-
+			while read line; do
+				if [[ $line =~ PORTAGE_TMPDIR= ]]; then
+					declare $line
+					temp="${PORTAGE_TMPDIR%\"}"
+					path="${temp#\"}"
+					_rcd-set_repopath_add_if_unique
+					found_default="true"
+					break
+				fi
+			done < /etc/portage/make.conf || die "failed to read make.conf"
```

Comment:
PORTAGE_TMPDIR is one of the bigger annoyance. Trying to read make.conf is more of a hack, and:
1. make.conf can be a directory, and have other rules like "source another.conf"
2. package.env may override PORTAGE_TMPDIR per package and then we just don't know

For 1. *could* use portageq to get the value, but portageq has meh startup time which is not great for `cd`
For 2, don't think there's much that can reasonably be done (automatically anyway)

An option could be to just use /var/tmp/portage by default and let the search paths be configurable (users could add extra paths for everything they use).

This aside, not sure if we really need an over-engineered tool (aka things in this project) to cd to workdir, ultimately just cd'ing to the latest timestamp directory in the tmpdir is about all that sounds useful.

Reply:
Thanks for your fast reply on this.

1. Yes I was thinking about using portageq instead, but refrained from it because of the startup time. 
If PORTAGE_TMPDIR isn't to be found in make.conf, we could just assume it's /var/tmp/portage. Users can set --path to something different in this case or put it in the config (once that's ready)
2. If that's the case users can still provide --path. Another option would be to fiddle with the output of emerge --info or something, but that is too slow again. I want this to be snappy and frictionless. 

Idk if people really need this. The question if we have something like this came up in the German gentoo Linux meetup. When fiddling with ebuilds (e.g. writing patches) I use to cd into WORKDIR all the time.
I would love to integrate the thing you said, so when you run workdir-cd without a package argument it just goes to the latest work directory in the tmpdir. This would allow for even faster operation. 

So if you don't want to have it I'm gonna shamelessly steal your script as a skeleton. I was hoping to get some input from this PR though, because mit bash-fu isn't at good as yours